File: python/web_tutorial_mastercopy/tut_worldmap_p1.py
from bridges.bridges import *
from bridges.data_src_dependent import *
from bridges.dl_element import *
from bridges.world_map import *

def main():
    # create the Bridges object, set credentials
#if TESTING
    # command line args provide credentials and server to test on
    args = sys.argv[1:]
    bridges = Bridges(int(args[0]), args[1], args[2])
    if len(args) > 3:
        bridges.connector.set_server(args[3])
#else
    bridges = Bridges(YOUR_ASSSIGNMENT_NUMBER, "YOUR_USER_ID", "YOUR_API_KEY")
#endif

    # set title, description
    bridges.set_title("World Map Using BRIDGES")
    bridges.set_description("Draws all countries and their boundaries in a default color")

    # with no arguments to the function, gets all countries of the world
    country_data = get_world_map_data()
    wm = WorldMap(country_data)

    #visualizes world map
    bridges.set_data_structure(wm)
    bridges.visualize()

if __name__ == '__main__':
    main()

File: c++/web_tutorial_mastercopy/tut_worldmap_p1.cpp
#include <string>

using namespace std;

#include "Bridges.h"
#include "DataSource.h"
#include "WorldMap.h"

using namespace bridges;

int main(int argc, char **argv) {

	// create bridges object
#if TESTING
	// command line args provide credentials and server to test on
	Bridges bridges (atoi(argv[1]), argv[2], argv[3]);
	if (argc > 4)
		bridges.setServer(argv[4]);
#else
	Bridges bridges (YOUR_ASSSIGNMENT_NUMBER, "YOUR_USER_ID", "YOUR_API_KEY");
#endif

	// set title
	bridges.setTitle("Drawing a World Map");

	// set description
	bridges.setDescription("This example illustrates how to draw a world map of all countries");

	// create a data source
	DataSource ds(&bridges);

	// the following call without arguments will get the data for all countries. Alternately
	// you can specify that with an argument such {"all"}.
	// This construct allow for styling the countries
	// vector<Country>  wm_data = ds.getWorldMapData();

	// declare the world map object with the map data	
	WorldMap wm;

	bridges.setMap(wm);
	
	// visualize
	bridges.setDataStructure(&wm);
	bridges.visualize();
	
	return 0;
}
